storage/src/main/java/org/apache/kafka/server/log/remote/metadata/storage/TopicBasedRemoteLogMetadataManager.java

```diff
@@ -377,7 +377,9 @@ private void initializeResources() {
         final NewTopic remoteLogMetadataTopicRequest = createRemoteLogMetadataTopicRequest();
         boolean topicCreated = false;
         long startTimeMs = time.milliseconds();
-        try (AdminClient adminClient = AdminClient.create(rlmmConfig.commonProperties())) {
```

Comment:
If the user configures the admin client incorrectly (eg: forgot to supply `remote.log.metadata.manager.listener.name`), then the thread will die without throwing any exception/log so moved the `Admin` instance creation within the try-block.

Reply:
Nice fix!